velox/exec/tests/WindowTest.cpp

```diff
@@ -79,6 +79,74 @@ TEST_F(WindowTest, spill) {
   ASSERT_GT(stats.spilledPartitions, 0);
 }
 
+TEST_F(WindowTest, rankLikeWithEqualValue) {
+  auto data = makeRowVector(
+      {"c1"},
+      {makeFlatVector<int64_t>(std::vector<int64_t>{1, 1, 1, 1, 1, 2, 2})});
+
+  createDuckDbTable({data});
+
+  const std::vector<std::string> kClauses = {
+      "sum(c1) over (order by c1 rows unbounded preceding)"};
+  core::PlanNodeId windowId;
+  auto plan = PlanBuilder()
+                  .values({data})
+                  .orderBy({"c1"}, false)
+                  .streamingWindow(kClauses)
+                  .capturePlanNodeId(windowId)
+                  .planNode();
+
+  auto spillDirectory = TempDirectoryPath::create();
+  auto task =
+      AssertQueryBuilder(plan, duckDbQueryRunner_)
+          .config(core::QueryConfig::kPreferredOutputBatchBytes, "1024")
+          .config(core::QueryConfig::kPreferredOutputBatchRows, "2")
+          .config(core::QueryConfig::kMaxOutputBatchRows, "2")
+          .config(core::QueryConfig::kSpillEnabled, "true")
+          .config(core::QueryConfig::kWindowSpillEnabled, "true")
+          .spillDirectory(spillDirectory->getPath())
+          .assertResults(
+              "SELECT *, sum(c1) over (order by c1 rows unbounded preceding) FROM tmp");
+}
+
+TEST_F(WindowTest, rankLikeOptimization) {
+  const vector_size_t size = 1'0;
+  auto data = makeRowVector(
+      {"d", "p", "s"},
+      {
+          // Payload.
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          // Partition key.
+          makeFlatVector<int16_t>(size, [](auto row) { return row % 2; }),
+          // Sorting key.
+          makeFlatVector<int32_t>(size, [](auto row) { return row; }),
+      });
+
+  createDuckDbTable({data});
+
+  const std::vector<std::string> kClauses = {
+      "rank() over (partition by p order by s)",
+      "row_number() over (partition by p order by s)",
+      "sum(d) over (partition by p order by s)"};
+  core::PlanNodeId windowId;
+  auto plan = PlanBuilder()
```

Comment:
Can there be a way to guarantee that we always use the RowStreamingWindowBuild in the test ?

Reply:
@aditi-pandit I tried to move the supportStreaming method into the WindowNode constructor within PlanNode. However, the compilation failed due to `velox_exec `already depending on `velox_core`. Consequently, `velox_core `must not depend on `velox_exec `to prevent a cyclic dependency. Would you have any suggestions?

Reply:
@JkSelf : That indeed is a problem.

Maybe add a QueryConfig to useRowStreaming in the Window operator, and use it in the test.

Reply:
@aditi-pandit Added a config kRowsStreamingWindow for test.

Reply:
@JkSelf @aditi-pandit I'm not sure I understand why we need this test-only configuration property. Would you clarify? I hope we can avoid adding this configuration property.

Reply:
@mbasmanova Ensure that this test conver the code path of RowStreamingWindow. However, coverage can still be attained without this configuration since this [condition ](https://github.com/facebookincubator/velox/pull/9025/files#diff-20dbd8f83c46e8d4bfcb760d98ddfcc75434651eb8fc0f109056ac468ff4ace6R46)holds true. I am fine to removing this configuration. 
@aditi-pandit Do you have any input?

Reply:
@mbasmanova : Since RowStreamingWindowBuild is used under certain conditions of streaming window, it is hard for us to test if RowStreamingWIndowBuild is actually used in the test. The test could default to regular StreamingWindowBuild and might not catch issues in RowStreamingWindowBuild. The config forces the logic to use RowStreamingWindowBuild.

Another option could be to have a special PlanBuilder method for forcing RowStreamingWindowBuild. But that would mean adding some fields to the WindowNode to separate Streaming and RowStreaming WindowBuild. So then we settled on config to trigger it.

@JkSelf : Any other thoughts ?

Reply:
> Another option could be to have a special PlanBuilder method for forcing RowStreamingWindowBuild. But that would mean adding some fields to the WindowNode to separate Streaming and RowStreaming WindowBuild. So then we settled on config to trigger it.
> 
> @JkSelf : Any other thoughts ?

@aditi-pandit @mbasmanova This option cannot be implemented due to a cyclic dependency [issue](https://github.com/facebookincubator/velox/pull/9025#discussion_r1616470742).

Reply:
Ideally, we would have a test that verifies the desired behavior of the new code and fails without it. For example, we could devise a query that OOMs if row-streaming path is not taken or if it doesn't work as expected. 

Alternatively, we can (1) use TestValue to ensure a code path is taken; (2) use runtime stats to indicate a certain path taken (similar to abandonedPartialAggregation in aggregation). We can also check memory usage as we expect row-streaming to use much less memory then regular streaming.

Reply:
@mbasmanova @aditi-pandit Use the TestValue to ensure  conver the RowStreamingWindow code path.